servers/services/src/main/java/org/projectnessie/services/impl/NamespaceApiImpl.java

```diff
@@ -94,9 +95,13 @@ public Namespace createNamespace(NamespaceParams params, Namespace namespace)
       Preconditions.checkArgument(!namespace.isEmpty(), "Namespace name must not be empty");
 
       Put put = Put.of(ContentKey.of(namespace.getElements()), namespace);
-      commit(branch, "create namespace " + namespace.name(), TreeApiImpl.toOp(put), validator);
+      Hash hash =
+          commit(branch, "create namespace " + namespace.name(), TreeApiImpl.toOp(put), validator);
 
-      return namespace;
+      Namespace namespaceWithId =
+          (Namespace) getStore().getValue(hash, Key.of(namespace.getElements()));
```

Comment:
Could we perhaps generify `getValue` to be `<T extends Content> T getValue(...)` to avoid trivial casts?

Reply:
It's only a special case for Namespace. Added a check.